desktop/mac/permission-requirements.md

```diff
@@ -7,34 +7,94 @@ redirect_from:
+The rejection is recorded for future runs to avoid prompting the user again.
+For any failure to ensure binaries are on the user's PATH, the user may need to manually add to their PATH the `/Applications/Docker.app/Contents/Resources/bin` or use the full path to docker binaries.
+
+A particular case is the installation of the `/var/run/docker.sock` symlink. Creating this symlink ensures various docker clients relying on the default docker socket path to work without additional changes. As the `/var/run` is mounted as a tmpfs, its content is deleted on restart, symlink to docker socket included.
+To ensure the docker socket exists after restart, Docker Desktop sets up a `launchd` startup task that creates the symlink by running `ln -s -f /Users/<user>/.docker/run/docker.sock /var/run/docker.sock`. This ensures, the user is not prompted all the time on startup to create the symlink. If the user rejects the prompt, the symlink and the startup task are not created and the user may have to explicitly set the DOCKER_HOST to `/Users/<user>/.docker/run/docker.sock` in the clients it is using. The docker CLI relies on the current context to retrieve the socket path, the current context is set to `desktop-linux` on Docker Desktop startup.
+
+### Binding privileged ports
+
+When running a container that requires privileged ports binding, Docker Desktop first attempts to bind it directly as an unprivileged process. In case of failure due to the OS preventing it, Docker Desktop checks if the `com.docker.vmnetd` privileged helper process is running to bind the privileged port through it. If the privileged helper process is not running, Docker Desktop prompts the user for authorization to run it under [launchd](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPSystemStartup/Chapters/CreatingLaunchdJobs.html).
+This configures the privileged helper to run as in the versions of Docker Desktop prior to v4.15. However, the functionality provided by this privileged helper has been stripped down. It now only supports port binding and caching the Registry Access Management policy.
+If the user declines the launch of the privileged helper process, the privileged port binding cannot be done and the docker CLI returns an error:
+```console
+$ docker run -p 127.0.0.1:80:80 docker/getting-started
+
+docker: Error response from daemon: Ports are not available: exposing port
+TCP 127.0.0.1:80 -> 0.0.0.0:0: failed to connect to /var/run/com.docker.vmnetd.sock:
+is vmnetd running?: dial unix /var/run/com.docker.vmnetd.sock: connect: connection
+refused.
+ERRO[0003] error waiting for container: context canceled
+```
+
+> **Note**
+>
+> The command may fail with the same error if user takes too long to authorize the prompt to start the helper process as it may timeout.
+
+The removal of the privileged helper process can be done in the typical way of removing launchd processes.
```

Comment:
Shall we move these instructions about removal to the `Privileged Helper` section?